codegen/table.go

```diff
@@ -1,51 +1,149 @@
 package codegen
 
 import (
+	"embed"
+	"fmt"
+	"io"
 	"reflect"
+	"text/template"
+	"unicode"
 
 	"github.com/cloudquery/plugin-sdk/schema"
 )
 
-type ResourceDefinition struct {
-	Name  string
-	Table *TableDefinition
-}
-
 type TableDefinition struct {
 	Name        string
 	Columns     ColumnDefinitions
 	Description string
 	Relations   []string
 
-	Resolver                      string
-	IgnoreError                   string
-	Multiplex                     string
-	PostResourceResolver          string
-	PreResourceResolver           string
+	Resolver             string
+	IgnoreError          string
+	Multiplex            string
+	PostResourceResolver string
+	PreResourceResolver  string
+
 	nameTransformer               func(reflect.StructField) string
 	skipFields                    []string
 	extraColumns                  ColumnDefinitions
 	structFieldsToUnwrap          []string
 	unwrapAllEmbeddedStructFields bool
+	customTypeResolvers           map[schema.ValueType]string // value = ResolverProvider
 }
 
-type ColumnDefinitions []ColumnDefinition
-
-type ColumnDefinition struct {
-	// Name name of the column
-	Name          string
-	Type          schema.ValueType
-	Resolver      string
-	Description   string
-	IgnoreInTests bool
-	Options       schema.ColumnCreationOptions
+func (t *TableDefinition) shouldUnwrapField(field reflect.StructField) bool {
+	return isFieldStruct(field.Type) && (t.unwrapAllEmbeddedStructFields && field.Anonymous || sliceContains(t.structFieldsToUnwrap, field.Name))
 }
 
-func (c ColumnDefinitions) GetByName(name string) *ColumnDefinition {
-	for _, col := range c {
-		if col.Name == name {
-			return &col
+func (t *TableDefinition) getUnwrappedFields(field reflect.StructField) []reflect.StructField {
+	reflectType := field.Type
+	if reflectType.Kind() == reflect.Ptr {
+		reflectType = reflectType.Elem()
+	}
+
+	fields := make([]reflect.StructField, 0)
+	for i := 0; i < reflectType.NumField(); i++ {
+		sf := reflectType.Field(i)
+		if t.ignoreField(sf) {
+			continue
 		}
+
+		fields = append(fields, sf)
+	}
+	return fields
+}
+
+func (t *TableDefinition) ignoreField(field reflect.StructField) bool {
+	return len(field.Name) == 0 || unicode.IsLower(rune(field.Name[0])) || sliceContains(t.skipFields, field.Name)
+}
+
+func (t *TableDefinition) addColumnFromField(field reflect.StructField, parent *reflect.StructField) {
+	if t.ignoreField(field) {
+		return
+	}
+
+	columnType, err := valueToSchemaType(field.Type)
+	if err != nil {
+		fmt.Printf("skipping field %s on table %s, got err: %v\n", field.Name, t.Name, err)
+		return
+	}
+
+	path := field.Name
+	name := t.nameTransformer(field)
+	// skip field if there is no name
+	if name == "" {
+		return
+	}
+	if parent != nil {
+		path = parent.Name + `.` + path
+		name = t.nameTransformer(*parent) + "_" + name
+	}
+	resolver := `schema.PathResolver("` + path + `")`
+	resolverProvider, ok := t.customTypeResolvers[columnType]
```

Comment:
Should `customTypeResolvers` return the full resolver here? I would expect that if someone overrides a resolver, they override all of it (including the path)

Reply:
No, I think that only the func that takes a path will be more robust (allowing overrides for unwrapped fields as well)